psdisk/psdisk.c

```diff
@@ -117,199 +116,158 @@ static void psdisk_showPartsTable(void)
 
 	fseek(psdisk_common.file, 0L, SEEK_END);
 	printf("\n");
-	printf(BOLDWHITE "File system %s: %ld bytes\n"RESET, psdisk_common.fileName, ftell(psdisk_common.file));
-	printf("Memory size %u bytes\n", psdisk_common.mem.memSize);
-	printf("Units: sector %u bytes \n", psdisk_common.mem.sectorSize);
+	printf(BOLDWHITE "Partition table %s: %ld bytes\n"RESET, psdisk_common.fileName, ftell(psdisk_common.file));
+	printf("Memory size: %u bytes\n", psdisk_common.memctx.memsz);
+	printf("Block size: %u bytes\n", psdisk_common.memctx.blksz);
+	printf("Sector size: %u bytes\n", psdisk_common.memctx.secsz);
 	printf("\n");
-	printf(BOLDWHITE PARTITION_HEADER_ALIGMENT RESET, "Name", "Start", "End", "Sectors", "Size", "Type");
+	printf(BOLDWHITE PARTITION_HEADER_ALIGMENT RESET, "Name", "Start", "End", "Blocks", "Size", "Type");
 	LIST_FOREACH(node, &psdisk_common.list, ptrs) {
-		printf(PARTITION_DATA_ALIGMENT, node->pHeader.name, node->pHeader.offset, node->pHeader.offset + node->pHeader.size,
-			   node->pHeader.size / psdisk_common.mem.sectorSize, node->pHeader.size, psdisk_getTypeName(node->pHeader.type));
+		printf(PARTITION_DATA_ALIGMENT, node->part.name, node->part.offset, node->part.offset + node->part.size,
+			   node->part.size / psdisk_common.memctx.blksz, node->part.size, psdisk_getTypeName(node->part.type));
 	}
 	printf("\n");
 }
 
 
 /* Callbacks to ptable library */
 
-static ssize_t psdisk_readImg(unsigned int addr, void *buff, size_t bufflen)
+static ssize_t psdisk_readImg(off_t offs, void *buff, size_t len)
 {
-	int offs;
-
-	if (PTABLE_WITH_OFFSET(psdisk_common.opts))
-		offs = addr;
-	else
-		offs = addr - (psdisk_common.mem.memSize - psdisk_common.mem.sectorSize);
-
-	if (fseek(psdisk_common.file, offs, SEEK_SET) != 0) {
-		fprintf(stderr,"Cannot change file position, err: %s.\n", strerror(errno));
+	if (fseek(psdisk_common.file, 0, SEEK_SET) != 0) {
```

Comment:
Did you mean?
```suggestion
	if (fseek(psdisk_common.file, offs, SEEK_SET) != 0) {
```

Reply:
The interface of `libptable` got changed again and `read()/write()` callbacks aren't used anymore.

However, `0` was correct. The assumption was that the whole partition table fits in a single sector and gets read with a single `read()` call. The `offs` would point to end of memory defined by `-m` option (as `liptable` expects to find the partition table there) but `psdisk` only generates partition table image file with starting offset 0.